code/mob/living/critter/ai/sawflyai.dm

```diff
@@ -78,30 +79,32 @@
 
 
 /datum/aiTask/timed/targeted/sawfly_attack/get_targets()
-	. = list()
+	. = list() // don't ask why the list is simply a period
 	var/mob/living/critter/robotic/sawfly/owncritter = holder.owner
 
-	for (var/mob/living/C in view(owncritter,target_range))
-		if(C == owncritter) continue
-
-		if(istype(C, /mob/living/critter/robotic/sawfly)) continue
+	for (var/mob/living/C in viewers(max_dist, owncritter))
 		if (C.health < -50 || !isalive(C)) continue
-		if (C.job == "Security Officer" || C.job == "Head of Security")
-			. = list(C) //found a secoff, just return that
+		if(istype(C, /mob/living/critter/robotic/sawfly)) continue
+
+		if(C.mind?.special_role)
+			if (istraitor(C) || isnukeop(C) || isspythief(C) || isnukeopgunbot(C)) // frens :)
+				if (!(C in owncritter.friends))
+					boutput(C, "<span class='alert'> [owncritter]'s IFF system silently flags you as an ally! </span>")
+					owncritter.friends += C
```

Comment:
This going to end up holding references to disposed mobs. It should use the weakref system, or just don't store refs to friends in the first place.